crd-generator/collector/src/main/java/io/fabric8/crd/generator/collector/CustomResourceCollector.java

```diff
@@ -0,0 +1,195 @@
+  }
+
+  public CustomResourceCollector withFileToScan(File... files) {
+    jandexCustomResourceClassScanner.withFileToScan(files);
+    return this;
+  }
+
+  public CustomResourceCollector withFilesToScan(Collection<File> files) {
+    jandexCustomResourceClassScanner.withFilesToScan(files);
+    return this;
+  }
+
+  public CustomResourceCollector withForceIndex(boolean forceIndex) {
+    jandexCustomResourceClassScanner.withForceIndex(forceIndex);
+    return this;
+  }
+
+  public CustomResourceCollector withForceScan(boolean forceScan) {
+    this.forceScan = forceScan;
+    return this;
+  }
+
+  public CustomResourceCollector withIncludePackages(Collection<String> packages) {
+    if (packages != null) {
+      packages.stream()
+          .filter(Objects::nonNull)
+          .map(pkg -> (Predicate<String>) s -> s.startsWith(pkg))
+          .reduce(Predicate::or)
+          .ifPresent(this.classNamePredicates::add);
+    }
+    return this;
+  }
+
+  public CustomResourceCollector withExcludePackages(Collection<String> packages) {
+    if (packages != null) {
+      packages.stream()
+          .filter(Objects::nonNull)
+          .map(pkg -> (Predicate<String>) s -> !s.startsWith(pkg))
+          .reduce(Predicate::or)
+          .ifPresent(this.classNamePredicates::add);
+    }
+    return this;
+  }
+
+  /**
+   * Find and load Custom Resource classes in the previously defined context.
+   * <p>
+   * If at least one Custom Resource class name is given, scanning for classes is skipped.
+   * Otherwise Custom Resource classes are searched in locations provided by
+   * {@link #withFileToScan(File...)} or {@link #withFilesToScan(Collection)}.
+   * </p>
+   *
+   * @return the Custom Resource classes
+   */
+  public List<Class<? extends HasMetadata>> findCustomResourceClasses() {
+    Set<String> customResourcesClassNames = new HashSet<>(customResourceClassNames);
+
+    if (forceScan || customResourcesClassNames.isEmpty()) {
+      // search only if custom resource class names are not explicitly given
+      customResourcesClassNames.addAll(jandexCustomResourceClassScanner.findCustomResourceClasses());
+      log.debug("Found {} custom resource classes", customResourcesClassNames.size());
+    } else {
+      log.debug("Using explicit {} custom resource classes and skip scanning", customResourcesClassNames.size());
+    }
+
+    Predicate<String> classNamePredicate = classNamePredicates.stream()
+        .reduce(Predicate::and)
+        .orElse(className -> true);
+
+    List<Class<? extends HasMetadata>> customResourceClasses = customResourcesClassNames.stream()
+        .filter(classNamePredicate)
+        .map(customResourceClassLoader::loadCustomResourceClass)
+        .collect(Collectors.toList());
+
+    log.debug("Found {} custom resource classes after filtering", customResourceClasses.size());
+    return customResourceClasses;
+  }
+
+  /**
+   * Resets all internal states to defaults.
+   */
+  public void reset() {
```

Comment:
Do we need this method to be public?

Reply:
Neither the maven plugin nor the CLI requires this method at the moment. But I thought it's a good indicator to communicate that the subcomponents of the CustomResourceCollector have an internal state and cache some things.

We can make this package-private and open it if someone really needs it.